src/natcap/invest/habitat_quality.py

```diff
@@ -1013,6 +913,78 @@ def _raster_values_in_bounds(raster_path_band, lower_bound, upper_bound):
     return values_valid
 
 
+def _decay_distance(dist_raster_path, max_dist, decay_type, target_path):
+    """Apply an exponential or linear decay to a distance transform raster.
+
+    The function will set pixles greater than ``max_dist`` to 0.
+
+    Args:
+        dist_raster_path (string): a filepath for the raster to decay.
+            The raster is expected to be a euclidean distance transform with
+            values meauring distance in pixels.
+        max_dist (float): max distance of threat in KM.
+        decay_type (string): a string defining which decay method to use.
+            Options include: 'linear' | 'exponential'.
+        target_path (string): a filepath for a float output raster.
+
+    Returns:
+        None
+    """
+    # get raster pixel size to determine how many pixels max_dist covers
+    threat_pixel_size = pygeoprocessing.get_raster_info(
+        dist_raster_path)['pixel_size']
+
+    # convert max distance (given in KM) to meters
+    max_dist_m = max_dist * 1000.0
+
+    # convert max distance from meters to the number of pixels that
+    # represents on the raster
+    max_dist_pixel = max_dist_m / abs(threat_pixel_size[0])
```

Comment:
Ah ok great, I see that these are already assumed to be square from the `align_and_resize_raster_stack` step.